builder/api/src/main/java/io/helidon/builder/api/Prototype.java

```diff
@@ -0,0 +1,414 @@
+                return (T) builder.buildPrototype();
+            }
+            if (instance instanceof io.helidon.common.Builder<?, ?> builder) {
+                return (T) builder.build();
+            }
+            return instance;
+        }
+    }
+
+    /**
+     * Extension of {@link io.helidon.builder.api.Prototype.Builder} that supports configuration.
+     * If a blueprint is marked as {@code @Configured}, build will accept configuration.
+     *
+     * @param <BUILDER>   type of the builder
+     * @param <PROTOTYPE> type of the prototype to be built
+     */
+    interface ConfiguredBuilder<BUILDER, PROTOTYPE> extends Builder<BUILDER, PROTOTYPE> {
+        /**
+         * Update builder from configuration.
+         *
+         * @param config configuration to use
+         * @return updated builder instance
+         */
+        BUILDER config(Config config);
+
+        /**
+         * Discover services from configuration.
+         *
+         * @param config               configuration located at the node of the service providers
+         *                             (either a list node, or object, where each child is one service)
+         * @param serviceLoader        helidon service loader for the expected type
+         * @param allFromServiceLoader whether all services from service loader should be used, or only the ones with configured
+         *                             node
+         * @param <S>                  type of the expected service
+         * @return list of discovered services
+         */
+        default <S extends NamedService, T extends ConfiguredProvider<S>> List<S>
+        discoverServices(Config config,
+                         HelidonServiceLoader<T> serviceLoader,
+                         Class<T> providerType,
+                         Class<S> configType,
+                         boolean allFromServiceLoader) {
+            return ProvidedUtil.discoverServices(config, serviceLoader, providerType, configType, allFromServiceLoader);
+        }
+    }
+
+    /**
+     * A prototype {@link Prototype.Blueprint} may extend this interface
+     * to explicitly reference the associated runtime type.
+     * <p>
+     * For example a {@code RetryPrototypeBlueprint} that extends a {@link Prototype.Factory}
+     * of {@code Retry}, will add methods to the prototype interface and builder to build an instance of
+     * {@code Retry} directly.
+     * A factory method must exist either on the runtime type (such as {@code Retry}) with signature
+     * {@code static Retry create(RetryPrototype)}, or on the prototype blueprint.
+     *
+     * @param <T> type of the runtime object (such as {@code Retry} in the description above)
+     */
+    interface Factory<T> {
+        /**
+         * Create a new instance of the runtime type from this config object.
+         *
+         * @return new configured runtime instance
+         */
+        T build();
+        /*
```

Comment:
address this comment.

Reply:
the comment is an explanatoin, not something to be addressed